### ISSUE TITLE
The pre-training pipeline of BLIP

### ISSUE THREAD

**Issue by wangjk666** [salesforce/BLIP#11]
Thanks a lot for your awesome work! I have some question about the pre-training pipeline of BLIP. Considering the application of Captioner and Filter to generate pretraining data, I wonder if the process for the complete training is: 
1. pre-training on noisy data
2. fine-tuning on COCO to obtain captioner and filter
3. using the captioner and filter from 2 to generate the final pretraining dataset
4. pre-training on the dataset obtained in 3

**Comment by LiJunnan1992** [comment]
Yes the process is exactly like you described!

**Comment by wangjk666** [comment]
Got it. Thanks a lot for your reply!

**Comment by wangjk666** [comment]
I have an additional question, will you train from scratch when you implement the 4th step (pre-training on the dataset obtained in 3)?

**Comment by LiJunnan1992** [comment]
Great question! Yes we pre-train from scratch, the ablation is shown in Table 13 of our paper.

**Comment by wangjk666** [comment]
Wooo, sorry for not noticing that. Thanks for your reply.

**Comment by rgtjf** [comment]
Great work! I notice that BLIP only adopts LM loss instead of MLM loss. Have you tried to train with LM loss and MLM loss at the same time, or is there any conflict between these two tasks?

**Comment by LiJunnan1992** [comment]
@rgtjf 
We haven't tried to combine MLM with LM. We choose LM to enable language decoding capabilities, and only used LM to keep things simple. My intuition is that adding MLM probably won't hurt.

**Comment by rgtjf** [comment]
Got it. Thanks for your quick reply!
